packages/backend-plugin-api/src/wiring/factories.ts

```diff
@@ -69,7 +69,16 @@ export interface BackendModuleConfig<TOptions> {
   ): void;
 }
 
-/** @public */
+/**
+ * @public
+ *
+ * Creates a new backend module for a given plugin.
+ *
+ * The `moduleId` should be equal to the module-specific prefix of the exported name, such
+ * that the full name is `<moduleId><PluginId>Module`.
+ *
+ * The `pluginId` should exactly match the `id` of the plugin that the module extends.
```

Comment:
I think this one warrants an extensive example, that includes the relevant module names and ids and variable names used. It's a bit hard to follow just the text here to make a naming decision, even for someone who's involved :)